doc/btrfs_in_partitioner.md

```diff
@@ -0,0 +1,326 @@
+one block device) would be visible as an entity on their own, instead of just
+some kind of property of the underlying block device.
+
+### Changes in the general list of devices
+
+To better reflect the interactions between the different devices and to make
+possible to visualize the multi-device filesystems, some changes would also be
+needed in the general lists of devices that are currently used all along the
+Partitioner.
+
+Let's take an example with several formatted partitions, a `bcache0` device
+based on `sda1` and `sda2`, an MD RAID formed by `sdb3` and `sdb7` and an LVM
+based on the `sdc` disk. That's all visualized in the following devices graph.
+
+![Example devicegraph](btrfs_in_partitioner/devicegraph.png)
+
+The classical list of the partitioner will display all those devices like the
+following picture.
+
+![Current full list of devices](btrfs_in_partitioner/current_full_list.png)
+
+With so many different kinds of possible rows (disks, partitions, RAIDs,
+LVM... and now bcache and multi-device filesystems), it's necessary to
+unify the "Type" and "FS Type" columns into a new one which makes a better
+job informing about the relationships of the different devices.
+
+![New full list of devices](btrfs_in_partitioner/better_full_list.png)
+
+With that change, multi-device filesystems could be represented as an extra row
+in the table of devices, with its corresponding block devices being labeled as
+"_Part of..._". The information about filesystem label and mount point will only
+be presented in the row corresponding to the multi-device filesystem, not in
+each of its individual block devices. See the following example in which the
+root filesystem spreads over `sdb2` and `sdb7`.
+
+![List including a multi-device Btrfs](btrfs_in_partitioner/multi_btrfs_in_list.png)
+
+The logic of the new column for each possible element is described in [this
+gist](https://gist.github.com/joseivanlopez/06e92d5784da7efad7a6feeeb8e2eac6).
+
+### Rethinking the dialogs to create/edit filesystems
+
+Other aspect that would need a better organization is the dialogs and workflows
+to format a device (or a set of devices, now that multi-device filesystems are
+possible) and to modify existing filesystems.
+
+Let's take a look at what would be the new workflow for creating and formating
+a new partition. It implies the same number of steps than in the classic
+workflow. But none of new steps mixes logic referred to the partition with logic
+related to the filesystem itself. That means filesystem-specific dialog(s) can
+be directly reused to create and edit multi-device filesystems.
+
+The first two screens remain identical. One for defining the size of the
+partition.
+
+![Partition creation step 1](btrfs_in_partitioner/new_part_1.png)
+
+And the role selection which allows to predefine some of the options for the
+following screen(s).
+
+![Partition creation step 2](btrfs_in_partitioner/new_part_2.png)
+
```

Comment:
I expected the "Role dialog" to be the third stop. Once I have defined my partition (size, encryption, id, etc) then I decide which role to apply (i.e., how I want to use my partition). In fact, if I decide to not format the partition, the role dialog is not needed at all, so we can finish the wizard just in the second step.

Reply:
This would also make the role "Raw partition" leftover.

Reply:
But the point of the "Role" screen is to prefill the options in the wizard before making any other decision. It's kind of weird to have questions about format/dontformat and about the Partition ID before selecting the role.

Reply:
I mean:

- Operating system: preselects "format", partition ID linux and the next available mount point
- Data: basically the same but with a different set of mount points
- Swap: preselects "format", the partition ID, the filesystem and the mount point
- EFI: preselects "format", the partition ID, the filesystem and the mount point
- Raw: preselect "don't format" and the LVM Partition ID.

If I select the role after having selected already whether to format and the partition ID... I don't know. It feels wrong.

Reply:
Well, if the partition role affects to the partition id, then I agree.